src/main/java/com/minecolonies/tileentities/TileEntityColonyBuilding.java

```diff
@@ -80,11 +80,20 @@ public void update()
 
         if (!worldObj.isRemote && colonyId == 0)
         {
-            Log.getLogger().fatal(String.format("TileEntityColonyBuilding at %s:[%d,%d,%d] has no colonyId",
-              worldObj.getWorldInfo().getWorldName(),
-              pos.getX(),
-              pos.getY(),
-              pos.getZ()));
+            Colony tempColony = ColonyManager.getColony(worldObj, this.getPosition());
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'tempColony' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)